File: jwj/week05/[2231] 분해합.py
'''
* Author    : Jang Woo Jin
* Date      : 2024.10.28(Mon)
* Runtime   : 2980 ms
* Memory    : 31120 KB
* Algorithm : Brute Force
'''

import sys
input = sys.stdin.readline

N = int(input())
lst = []

for n in range(N-1, 1, -1):
    n = list(str(n))
    tot = int(''.join(map(str, n)))
    for i in range(len(n)):
        tot += int(n[i])
    if tot == N:
        lst.append(''.join(map(str, n)))

if lst:
    lst.sort()
    print(lst[0])
else:
    print(0)

File: 이정훈/5주차/[2231] 분해합.cpp
/*
* Author    : Lee JeongHun
* Date      : 2024.11.02(Sat)
* Runtime   : 20 ms
* Memory    : 2020 KB
* Algorithm : brute force
*/

#include <iostream>
#include <string>

using namespace std;

int main()
{
    // 입력 처리
    int n;
    cin >> n;

    // 변수 선언
    int constructor = 0;
    
    // 생성자는 n 보다 작다
    for(int i = 0; i < n; ++i)
    {
        int sum = i;
        string target = to_string(i);
        for(char c : target)
        {
            sum += c - '0';
        }

        if(sum == n)
        {
            constructor = i;
            break;
        }
    }

    // 정답 출력
    cout << constructor;

    return 0;
}